### PR TITLE
Add `ChosenDeploymentAlphaLocatorFlow` tests

### PR DESCRIPTION
**Pull request by jamiebrynes7**
#### Description
Following on from https://github.com/spatialos/gdk-for-unity/pull/1108, added some tests for the `ChosenDeploymentAlphaLocatorFlow` too! 

Did a bit of tidy up and moved `ChosenDeploymentAlphaLocatorFlow` and `SessionConnectionFlowInitializer` to their own files instead of being inside the `ClientWorkerConnector`.

**Remaining work**
- [x] Run the tests in CI
- [x] Changelog

### CODE REVIEW

File: ci/test.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+echo "--- Testing Unity: Editmode :writing_hand:"
+
+pushd "workers/unity"
+    dotnet run -p "${PROJECT_DIR}/.shared-ci/tools/RunUnity/RunUnity.csproj" -- \
+        -batchmode \
+        -projectPath "${PROJECT_DIR}/workers/unity" \
+        -runTests \
+        -testPlatform editmode \
+        -logfile "${PROJECT_DIR}/logs/unity-editmode-test-run.log" \
+        -testResults "${EDITMODE_TEST_RESULTS_FILE}" \
```
Comment (zeroZshadow) [inline ci/test.sh]: `-editorTestsResultFile`

File: ci/test.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+EDITMODE_TEST_RESULTS_FILE="${PROJECT_DIR}/logs/editmode-test-results.xml"
+
+echo "--- Testing Unity: Editmode :writing_hand:"
+
+pushd "workers/unity"
+    dotnet run -p "${PROJECT_DIR}/.shared-ci/tools/RunUnity/RunUnity.csproj" -- \
+        -batchmode \
+        -projectPath "${PROJECT_DIR}/workers/unity" \
+        -runTests \
```
Comment (zeroZshadow) [inline ci/test.sh]: `-runEditorTests`
No need for the platform one anymore

File: workers/unity/Assets/Fps/Tests/EditmodeTests/ChosenDeploymentAlphaLocatorFlowTests.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    {
+        private SpatialdManager spatiald;
+        private Improbable.Worker.CInterop.Connection connection;
+
+        [OneTimeSetUp]
+        public void OneTimeSetup()
+        {
+            if (!DevAuthTokenUtils.TryGenerate())
+            {
+                throw new ArgumentException("Could not generate a dev auth token.");
```
Comment (zeroZshadow) [inline workers/unity/Assets/Fps/Tests/EditmodeTests/ChosenDeploymentAlphaLocatorFlowTests.cs]: Is this an actual argument exception?

File: workers/unity/Assets/Fps/Tests/EditmodeTests/ChosenDeploymentAlphaLocatorFlowTests.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        public void TearDown()
+        {
+            connection?.Dispose();
+            connection = null;
+        }
+
+        [Test]
+        public void CreateAsync_succeeds_if_there_is_matching_deployment()
+        {
+            using (var depl = spatiald.StartLocalDeployment("target_deployment", "default_launch.json").Result)
```
Comment (zeroZshadow) [inline workers/unity/Assets/Fps/Tests/EditmodeTests/ChosenDeploymentAlphaLocatorFlowTests.cs]: Move the `default_launch.json` to the top to make replacement easier later

File: workers/unity/Assets/Fps/Tests/EditmodeTests/ChosenDeploymentAlphaLocatorFlowTests.cs (hunk 3 above)
Comment (jamiebrynes7) [reply workers/unity/Assets/Fps/Tests/EditmodeTests/ChosenDeploymentAlphaLocatorFlowTests.cs]: Nah probably not, can be just an `Exception`

File: workers/unity/Assets/Fps/Tests/EditmodeTests/ChosenDeploymentAlphaLocatorFlowTests.cs (hunk 4 above)
Comment (jamiebrynes7) [reply workers/unity/Assets/Fps/Tests/EditmodeTests/ChosenDeploymentAlphaLocatorFlowTests.cs]: 👍

**Pull request merged**